### DISCUSSION TITLE
how do I import a video?

### DISCUSSION THREAD

**Discussion by Raj2032** [robertknight/ocrs#86] [Q&A] [answered]
I want to instead of trying to scan a still image I want to instead import a video file and detect all the text inside it and print it to the console maybe I could try and use opencv if this is possible? just not quite too sure how I would get it integrated into ocrs?

**Comment by robertknight** [comment, marked as answer]
To process a video you will need to:

1. Find a way to extract frames from the video as an image. I am not that familiar with the state of video processing in Rust, but a wrapper for GStreamer, FFmpeg or OpenCV might be a starting point (eg. the `VideoCapture` APIs)
2. Convert the image to 8-bit RGB, if not already in that format
3. Use [`ocrs::ImageSource::from_bytes`](https://docs.rs/ocrs/latest/ocrs/struct.ImageSource.html#method.from_bytes) to prepare the image for Ocrs to process.
4. Once you have an `ImageSource`, you can then feed it into text recognition following the steps in [the example](https://github.com/robertknight/ocrs/blob/fe6de19db01ea5aaf4a4e891e21c74bb58394277/ocrs/examples/hello_ocr.rs#L68).

To speed up extraction, you will probably want to sample only a subset of the frames. For example, extract only 1 in every N frames.

**Reply by Raj2032** [reply]
ok thanks mate :)
